module/move/willbe/src/endpoint/module_headers.rs

```diff
@@ -0,0 +1,219 @@
+			.and_then( | i | i.as_str() )
+			.and_then( | s | s.parse::< Stability >().ok() )
+			.unwrap_or( Stability::Experimental );
+
+			let repository_url = doc
+			.get( "package" )
+			.and_then( | metadata | metadata.get( "repository" ) )
+			.and_then( | url | url.as_str() )
+			.map( String::from )
+			.ok_or_else::< Error, _ >( || err!( "package.repository not found in module Cargo.toml" ) )?;
+
+			let module_name = doc
+			.get( "package" )
+			.and_then( | workspace  | workspace.get( "name" ) )
+			.and_then( | url | url.as_str() )
+			.map( String::from )
+			.ok_or_else::< Error, _ >( || err!( "master_branch not found in module Cargo.toml" ) )?;
+
+			let discord_url = doc
+			.get( "package" )
+			.and_then( | workspace  | workspace.get( "metadata" ) )
+			.and_then( | metadata | metadata.get( "discord_url" ) )
+			.and_then( | url | url.as_str() )
+			.map( String::from )
+			.or_else( || default_discord_url.clone() );
+
+			Ok
+			(
+				Self
+				{
+					stability,
+					module_name,
+					repository_url,
+					discord_url,
+				}
+			)
+		}
+
+		/// Convert `ModuleHeader`to header.
+		fn to_header( self ) -> Result< String >
+		{
+			let discord = if self.discord_url.is_some()
+			{
+				format!( "\n[![discord](https://img.shields.io/discord/872391416519737405?color=eee&logo=discord&logoColor=eee&label=ask)]({})", self.discord_url.unwrap() )
+			} 
+			else
+			{
+				"".into()
+			};
+			let repo_url = url::extract_repo_url( &self.repository_url ).and_then( | r | url::git_info_extract( &r ).ok() ).ok_or_else::< Error, _ >( || err!( "Fail to parse repository url" ) )?;
+			Ok( format!
+			(
+				"{}\
+				[![rust-status](https://github.com/{}/actions/workflows/Module{}Push.yml/badge.svg)](https://github.com/{}/actions/workflows/Module{}Push.yml)\
+				[![docs.rs](https://img.shields.io/docsrs/{}?color=e3e8f0&logo=docs.rs)](https://docs.rs/{})\
+				[![Open in Gitpod](https://raster.shields.io/static/v1?label=try&message=online&color=eee&logo=gitpod&logoColor=eee)](https://gitpod.io/#RUN_PATH=.,SAMPLE_FILE=sample%2Frust%2F{}_trivial_sample%2Fsrc%2Fmain.rs,RUN_POSTFIX=--example%20{}_trivial_sample/https://github.com/{}){}",
+				stability_generate( &self.stability ),
+				repo_url, self.module_name.to_case( Case::Pascal ), repo_url, self.module_name.to_case( Case::Pascal ),
+				self.module_name, self.module_name,
+				self.module_name, self.module_name, repo_url,
+				discord,
+			) )
+		}
+	}
+	
+	fn workspace_discord_url( path: &CargoTomlLocation ) -> Result< Option< String > >
+	{
+		if !path.exists()
+		{
+			bail!( "Cannot find Cargo.toml" )
+		}
+		let mut contents = String::new();
+
+		File::open( path )?.read_to_string( &mut contents )?;
+
+		let doc = contents.parse::< Document >()?;
+
+		let discord = doc
+		.get( "workspace" )
+		.and_then( | package | package.get( "metadata" ) )
+		.and_then( | metadata | metadata.get( "discord_url" ) )
+		.and_then( | i | i.as_str() )
+		.map( | s | s.to_string() );

```

Comment:
I see we have a similar situation here to the one I mentioned earlier regarding packages. In this section of our code, we're retrieving a discord_url from the workspace metadata. As these operations are closely related to workspace configuration, moving them to workspace.rs would help maintain consistency and improve the organization of our codebase.